src/H5FLprivate.h

```diff
@@ -27,7 +27,7 @@
 
 /* Macros for turning off free lists in the library */
 /*#define H5_NO_FREE_LISTS*/
-#if defined H5_NO_FREE_LISTS || defined H5_USING_MEMCHECKER
+#if defined H5_NO_FREE_LISTS || defined H5_USING_MEMCHECKER || defined H5_HAVE_MULTITHREAD
```

Comment:
This was the approach that I suggested, though @jrmainzer had some concerns about going that way, so we may want to consider something different.